modules/tensor_mechanics/src/materials/ComputeCappedWeakInclinedPlaneStress.C

```diff
@@ -0,0 +1,103 @@
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "ComputeCappedWeakInclinedPlaneStress.h"
+#include "RotationMatrix.h" // for rotVecToZ
+
+template<>
+InputParameters validParams<ComputeCappedWeakInclinedPlaneStress>()
+{
+  InputParameters params = validParams<ComputeCappedWeakPlaneStress>();
+  params.addClassDescription("Capped weak inclined-plane plasticity stress calculator");
+  params.addRequiredParam<RealVectorValue>("normal_vector", "The normal vector to the weak plane");
+  return params;
+}
+
+ComputeCappedWeakInclinedPlaneStress::ComputeCappedWeakInclinedPlaneStress(const InputParameters & parameters) :
+    ComputeCappedWeakPlaneStress(parameters),
+    _n_input(getParam<RealVectorValue>("normal_vector")),
+    _n(declareProperty<RealVectorValue>("weak_plane_normal")),
+    _n_old(declarePropertyOld<RealVectorValue>("weak_plane_normal")),
+    _rot(RealTensorValue())
+{
+  if (_n_input.norm() == 0)
+    mooseError("ComputeCappedWeakInclinedPlaneStress: normal_vector must not have zero length");
+  else
+    _n_input /= _n_input.norm();
+}
+
+void
+ComputeCappedWeakInclinedPlaneStress::initQpStatefulProperties()
+{
+  ComputeCappedWeakPlaneStress::initQpStatefulProperties();
+  _n[_qp] = _n_input;
+}
+
+void
+ComputeCappedWeakInclinedPlaneStress::computeQpStress()
+{
+  rotate();
+
+  if (_t_step >= 2)
+    _step_one = false;
+
+  ComputeCappedWeakPlaneStress::returnMap();
+
+  unrotate();
+
+  //Update measures of strain
+  _elastic_strain[_qp] = _elastic_strain_old[_qp] + _strain_increment[_qp] - (_plastic_strain[_qp] - _plastic_strain_old[_qp]);
+
+  //Rotate the tensors to the current configuration
+  if (_perform_finite_strain_rotations)
+  {
+    _stress[_qp] = _rotation_increment[_qp]*_stress[_qp]*_rotation_increment[_qp].transpose();
+    _elastic_strain[_qp] = _rotation_increment[_qp] * _elastic_strain[_qp] * _rotation_increment[_qp].transpose();
+    _plastic_strain[_qp] = _rotation_increment[_qp] * _plastic_strain[_qp] * _rotation_increment[_qp].transpose();
+
+    // Rotate n by _rotation_increment
+    for (unsigned int i = 0; i < LIBMESH_DIM; ++i)
+    {
+      _n[_qp](i) = 0.0;
+      for (unsigned int j = 0; j < LIBMESH_DIM; ++j)
+        _n[_qp](i) += _rotation_increment[_qp](i, j) * _n_old[_qp](j);
+    }
+  }
+}
+
+void
+ComputeCappedWeakInclinedPlaneStress::rotate()
+{
+  _rot = RotationMatrix::rotVecToZ(_n[_qp]);  // rotation matrix that will rotate _n to the "z" axis
+
+  // _elasticity_tensor and _strain_increment are both "const" as usually in computing
+  // stress these don't need to be altered.  Here, however, we need to rotate them.
+  (const_cast<MaterialProperty<RankFourTensor> &>(_elasticity_tensor))[_qp].rotate(_rot);
```

Comment:
Yes, you're right about the parent class.  I know const_cast is undesirable.  I'm half hoping that in #8316 (which is a complete refactor of the parent class) this might become unnecessary.  But i don't want to do the copy-and-rotate in this PR.